src/etl/extract/operations.py

```diff
@@ -0,0 +1,178 @@
+"""
+Extract-stage predefined operations.
+
+Every extract config file declares a sequence of operations for pulling columns
+of data out of a clinical file. Each operation is defined as a function that
+takes the original data as input and returns one or more columns.
+
+This module stores pre-made function wrappers that when invoked will return
+functions for performing common actions such as fetching a column and
+replacing its values with other values.
+
+See: docs/design/extract_config_format.py for function details
+"""
+from pandas import DataFrame
+from common.pandas_utils import get_col, safe_pandas_replace
+from common.type_safety import function, assert_safe_type
+
+
+def df_map(m):
+    """
+    For when you just can't figure out how to do what you want to do with the
+    other predefined operations. Take a whole dataframe and return a dataframe
+    with as many standard concept columns as you want.
+
+    Wraps the dataframe mapping operation in a function that takes only a
+    dataframe as input and returns a new dataframe.
+
+    :param m: A function to apply to a DataFrame
+    :returns: A function that applies the specified m operation
+    """
+    assert_safe_type(m, function)
+
+    def df_map_func(df):
+        new_df = df.copy()
+        return m(new_df)
```

Comment:
We should probably do an assert safe type on the output of m. It should be a dataframe

Reply:
you did it! 🙆‍♂️